java-manta-client/src/main/java/com/joyent/manta/client/crypto/AesCtrCipherDetails.java

```diff
@@ -64,60 +64,57 @@ public boolean plaintextSizeCalculationIsAnEstimate() {
-                    startInclusive);
+            String msg = "Start position must be zero or higher (startInclusive=" + startInclusive + ")";
             throw new IllegalArgumentException(msg);
         }
 
-        if (startInclusive > plaintextMax) {
-            String msg = String.format("Start position must be less than maximum "
-                            + "ciphertext size [%d]. Actually: %d",
-                    plaintextMax, startInclusive);
+        if (startInclusive >= plaintextMax) {
+            String msg = "Start position must be less than maximum plaintext size (startInclusive="
+                    + startInclusive + ", plaintextMax=" + plaintextMax + ")";
             throw new IllegalArgumentException(msg);
         }
 
         if (endInclusive < 0) {
-            String msg = String.format("End position must be zero or higher. Actually: %d",
-                    endInclusive);
+            String msg = "End position must be zero or higher (endInclusive=" + endInclusive + ")";
             throw new IllegalArgumentException(msg);
         }
 
-        if (endInclusive > plaintextMax) {
-            String msg = String.format("End position must be less than maximum "
-                            + "ciphertext size [%d]. Actually: %d",
-                    plaintextMax, endInclusive);
+        if (endInclusive >= plaintextMax) {
+            String msg = "End position must be less than maximum plaintext size (endInclusive=" + endInclusive
+                    + ", plaintextMax=" + plaintextMax + ")";
             throw new IllegalArgumentException(msg);
         }
 
         final int blockSize = getBlockSizeInBytes();
-        long adjustedStart;
-        long adjustedEnd;
-        long plaintextEndLength;
 
-        // How many bytes do we offset in the first block
-        long plaintextStartAdjustment = startInclusive % blockSize;
+        // How many bytes do we offset in the first block?
+        final long plaintextBytesToSkipInitially = startInclusive % blockSize;
 
-        // The first block is considered block 0
-        double startBlockNumber = Math.floor(startInclusive / blockSize);
+        // Get the ciphertext block index for the start byte (zero based)
+        final long startingBlockNumberInclusive = startInclusive / blockSize;
 
-        // Content-Range: bytes=adjustedStart-adjustedEnd
-        adjustedStart = ((long) startBlockNumber) * blockSize;
+        // Get the ciphertext byte position for the start block (block size bytes per block)
+        final long ciphertextStartPositionInclusive = startingBlockNumberInclusive * blockSize;
 
-        // Get the full block for the end range, plus an additional block
-        double endBlockNumber = Math.ceil(endInclusive / blockSize) + 1;
+        // Get the ciphertext block index for the end byte (zero based)
+        final long endingBlockNumberInclusive = endInclusive / blockSize;
```

Comment:
I designed the MantaEncryptedObjectInputStream to bound on the amount of plaintext bytes read in order to deal with any problems calculating ciphertext positions of ranges. For some algorithms and modes, we can never get the exact byte to end on. I'm trying to understand where above the off by one error is. Can you point this out?